Cleans up the fallout from plane cube

LemonInTheDark opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Alright.
Makes cleaning bubbles respect planes
Adds support for updating overlays on move, fixing an issue with pointing at items Adds better error messages for failing to provide args for mutable_appearance() Fixes a bug where string overlays were not respecting insertion order
ensures fullscreen objects START offset, so things are always proper
ensures chatmessages always have the right offset
whoops, the above lighting plane should actually be ABOVE the lighting plane
cleans up the fire overlay a tad
makes the sparks overlay non global
Adds a unit test for plane masters that are shrunk by multiz being double shrunk
This is slightly hacky because of how I'm handing the plane master group, but it's not THAT bad, and gives me some real good coverage
Properly targets the seethrough plane at the game world plate. This fixes unit tests, and also just makes more sense
Ensures camera chunks index at 1 instead of 0. Also removes the restriction on powers of 2 chunk size, cause that's dumb

Closes #70202
Closes #70195
Fixes #70515
Fixes #70329

LemonInTheDark committed:
Cleans up the fallout from plane cube

Alright.
Makes cleaning bubbles respect planes
Adds support for updating overlays on move, fixing an issue with pointing at items
Adds better error messages for failing to provide args for mutable_appearance()
Fixes a bug where string overlays were not respecting insertion order

Mothblocks commented on code/datums/mutable_appearance.dm:
```diff
@@ -18,14 +18,17 @@
 		else
 			stack_trace("No plane offset passed in as context for a non floating mutable appearance, ya done fucked up")
+	else if(!isnull(offset_spokesman) && !isatom(offset_spokesman))
+		stack_trace("Why did you pass in offset_spokesman as [offset_spokesman]?")
```
you dont even document what offset_spokesman *is*. i was trying to use this code the other day and had no idea what on earth it was. give this an actually useful stack trace AND have proper documentation

LemonInTheDark replied:
:( sorry

LemonInTheDark committed:
Adds documentation for offset spokesman and offset_const

LemonInTheDark committed:
Better stack trace

Mothblocks commented on code/controllers/subsystem/mapping.dm:
```diff
@@ -768,6 +768,8 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 		GLOB.cryo_overlays_cover_on += create_cryo_overlay(offset, "cover-on")
 		GLOB.cryo_overlays_cover_off += create_cryo_overlay(offset, "cover-off")
+		GLOB.cleaning_bubbles_lower += generate_bubbles(GAME_PLANE, offset)
+		GLOB.cleaning_bubbles_higher += generate_bubbles(ABOVE_GAME_PLANE, offset)
```
I know this was from the other PR but what is this whole proc? This looks like the kind of thing that is going to trip feature coders up, that they'll have to edit the mapping subsystem to get...something?

LemonInTheDark replied:
I wish there was a better way of handling it.
Basically, people sometimes want to store global versions of mutable appearances (though in all honesty this is hardly needed most of the time)
We need to generate a new mutable appearance each time, because they tend to involve planes. So the list gets indexed by the offset to use, and we generate a new MA each time somethin increases the offset.

Lemme do a pass for dumb uses of this

LemonInTheDark replied:
The only reason to do it at this scale is like, KINDA BARELY cpu time things, but it's like, nothing.
If we really cared we could setup a proc to autocache these, and assert something about their mutability afterwards.

LemonInTheDark committed:
Removes some redundant uses of cached MAs

At this scale, attempting to cache MAs like this has 0 impact on anything
And just makes things more messy then they need to be

LemonInTheDark committed:
ensures fullscreen objects START offset, so things are always proper

LemonInTheDark committed:
ensures chatmessages always have the right offset

LemonInTheDark committed:
fixes compile

LemonInTheDark committed:
whoops, the above lighting plane should actually be ABOVE the lighting plane

LemonInTheDark committed:
fixes compile, also cleans up the fire overlay a tad

LemonInTheDark committed:
Adds a unit test for plane masters that are shrunk by multiz being double shrunk

This is slightly hacky because of how I'm handing the plane master
group, but it's not THAT bad, and gives me some real good coverage

LemonInTheDark committed:
Properly targets the seethrough plane at the game world plate. This fixes unit tests, and also just makes more sense

LemonInTheDark commented:
what did I do to deserve this

optimumtact commented:
potatoe

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into cube-fix

LemonInTheDark commented:
god damn it why doesn't it compile

LemonInTheDark committed:
whoops

LemonInTheDark commented:
![image](https://user-images.githubusercontent.com/58055496/195812101-db03ac7e-3063-46a5-b259-894e294ba959.png)
NOOOOOOOO

LemonInTheDark commented:
Oh

LemonInTheDark committed:
oh

LemonInTheDark committed:
adds datum support for allocate(), cleans up a harddel from testing

LemonInTheDark committed:
Makes camera chunks index at 1, and also makes them support non powers of two sizes, since that was unneeded

LemonInTheDark committed:
fixes runtime in allocate

Mothblocks reviewed (approved):
This is OK after some test merges
